http/action/src/test/java/io/knotx/databridge/http/action/HttpActionNodeLogTest.java

```diff
@@ -0,0 +1,272 @@
+import java.util.function.Consumer;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+@ExtendWith(VertxExtension.class)
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.WARN)
+public class HttpActionNodeLogTest {
+
+  private static final String ACTION_ALIAS = "httpAction";
+  private static final String RAW_BODY = "<html>Hello</html>";
+  private static final String APPLICATION_JSON = "application/json";
+  private static final String APPLICATION_TEXT = "application/text";
+  private static final String JSON = "JSON";
+  private static final String JSON_BODY = "{\n" +
+      "  \"id\": 21762532,\n" +
+      "  \"url\": \"http://knotx.io\",\n" +
+      "  \"label\": \"Product\"\n" +
+      "}";
+  private static final JsonObject EMPTY_JSON = new JsonObject();
+  private WireMockServer wireMockServer;
+
+  static Stream<Arguments> succeedingRequestsCasesWithLogLevels() {
+    return Stream.of(//Content-Type, forceJson, JSON predicate, Body, LogLevel
+        Arguments.of(APPLICATION_JSON, false, null, JSON_BODY, ActionLogLevel.INFO),
+        Arguments.of(APPLICATION_TEXT, true, null, JSON_BODY, ActionLogLevel.INFO),
+        Arguments.of(APPLICATION_JSON, false, JSON, JSON_BODY, ActionLogLevel.INFO),
+        Arguments.of(APPLICATION_JSON, false, null, JSON_BODY, ActionLogLevel.ERROR),
+        Arguments.of(APPLICATION_TEXT, true, null, JSON_BODY, ActionLogLevel.ERROR),
+        Arguments.of(APPLICATION_JSON, false, JSON, JSON_BODY, ActionLogLevel.ERROR)
+    );
+  }
+
+  static Stream<Arguments> failingBodyDecodingCasesWithLogLevels() {
+    return Stream.of(//Content-Type, forceJson, JSON predicate, Body, LogLevel
+        Arguments.of(APPLICATION_JSON, false, null, RAW_BODY, ActionLogLevel.INFO),
+        Arguments.of(APPLICATION_TEXT, true, null, RAW_BODY, ActionLogLevel.INFO),
+        Arguments.of(APPLICATION_JSON, false, JSON, RAW_BODY, ActionLogLevel.INFO),
+        Arguments.of(APPLICATION_JSON, false, null, RAW_BODY, ActionLogLevel.ERROR),
+        Arguments.of(APPLICATION_TEXT, true, null, RAW_BODY, ActionLogLevel.ERROR),
+        Arguments.of(APPLICATION_JSON, false, JSON, RAW_BODY, ActionLogLevel.ERROR)
+    );
+  }
+
+  static Stream<Arguments> failingRequestsCasesWithLogLevels() {
+    return Stream.of(//Content-Type, forceJson, JSON predicate, Body, LogLevel
+        Arguments.of(APPLICATION_TEXT, false, JSON, JSON_BODY, ActionLogLevel.INFO),
+        Arguments.of(APPLICATION_TEXT, true, JSON, JSON_BODY, ActionLogLevel.INFO),
```

Comment:
Web Client throws an exception because of `JSON` predicate. The `forceJson` has no effect in this case. We can update this variable name to something like:
`webClientExceptionWithLogLevels`